packages/catalog-process/test/utils.ts

```diff
@@ -0,0 +1,196 @@
+  eservices: EServiceCollection
+): Promise<void> => {
+  await eservices.insertOne({
+    data: eService,
+    metadata: {
+      version: 0,
+    },
+  });
+};
+
+export const writeAgreementInReadmodel = async (
+  agreement: Agreement,
+  agreements: AgreementCollection
+): Promise<void> => {
+  await agreements.insertOne({
+    data: agreement,
+    metadata: {
+      version: 0,
+    },
+  });
+};
+
+export const writeTenantInReadmodel = async (
+  tenant: Tenant,
+  tenants: TenantCollection
+): Promise<void> => {
+  await tenants.insertOne({
+    data: tenant,
+    metadata: {
+      version: 0,
+    },
+  });
+};
+
+export const getMockAuthData = (organizationId?: string): AuthData => ({
+  organizationId: organizationId || uuidv4(),
+  userId: uuidv4(),
+  userRoles: [],
+  externalId: {
+    value: "123456",
+    origin: "IPA",
+  },
+});
+
+export const buildDescriptorSeed = (
+  descriptor: Descriptor
+): EServiceDescriptorSeed => ({
+  audience: descriptor.audience,
+  voucherLifespan: descriptor.voucherLifespan,
+  dailyCallsPerConsumer: descriptor.dailyCallsPerConsumer,
+  dailyCallsTotal: descriptor.dailyCallsTotal,
+  agreementApprovalPolicy: "AUTOMATIC",
+  description: descriptor.description,
+  attributes: {
+    certified: [],
+    declared: [],
+    verified: [],
+  },
+});
+
+export const getMockEService = (): EService => ({
+  id: uuidv4(),
+  name: "eService name",
+  description: "eService description",
+  createdAt: new Date(),
+  producerId: uuidv4(),
+  technology: technology.rest,
+  descriptors: [],
+  attributes: undefined,
+});
+
+export const getMockDescriptor = (): Descriptor => ({
+  id: uuidv4(),
+  version: "0",
+  docs: [],
+  state: descriptorState.draft,
+  audience: [],
+  voucherLifespan: 60,
+  dailyCallsPerConsumer: 10,
+  dailyCallsTotal: 1000,
+  createdAt: new Date(),
+  serverUrls: [],
+  agreementApprovalPolicy: "Automatic",
+  attributes: {
+    certified: [],
+    verified: [],
+    declared: [],
+  },
+});
+
+export const getMockTenant = (): Tenant => ({
+  name: "A tenant",
+  id: uuidv4(),
+  createdAt: new Date(),
+  attributes: [],
+  externalId: {
+    value: "123456",
+    origin: "IPA",
+  },
+  features: [],
+  mails: [],
+});
+
+export const getMockAgreement = ({
+  eServiceId,
+  descriptorId,
+  producerId,
+  consumerId,
+}: {
+  eServiceId: string;
+  descriptorId: string;
+  producerId: string;
+  consumerId: string;
+}): Agreement => ({
+  id: uuidv4(),
+  createdAt: new Date(),
+  eserviceId: eServiceId,
+  descriptorId,
+  producerId,
+  consumerId,
+  state: agreementState.active,
+  verifiedAttributes: [],
+  certifiedAttributes: [],
+  declaredAttributes: [],
+  consumerDocuments: [],
+  stamps: {
+    submission: undefined,
+    activation: undefined,
+    rejection: undefined,
+    suspensionByProducer: undefined,
+    suspensionByConsumer: undefined,
+    upgrade: undefined,
+    archiving: undefined,
+  },
+});
+
+export function decode<I extends object>({
```

Comment:
`decode` name may be a little bit to general. I would be more specific calling it `decodeProtobufPayload` or something like that

Reply:
Updated